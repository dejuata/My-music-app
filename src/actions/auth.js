import {
    getAuth,
    signInWithPopup,
    createUserWithEmailAndPassword,
    updateProfile,
    signInWithEmailAndPassword,
    signOut
} from 'firebase/auth';
import { googleAuthProvider } from '../firebase/firebaseConfig';
import { types } from "../types/types";
import { finishLoading, startLoading } from './ui';
import Swal from 'sweetalert2'


export const startLoginEmailPassword = (email, password) =>{

    return (dispatch) =>{

        dispatch( startLoading() );

        const auth = getAuth();
        signInWithEmailAndPassword(auth, email, password)
            .then(({ user }) => {
                console.log('dispatch login')
                dispatch(login(user.uid, user.displayName, 'firebase', ''))
            })
            .catch( (err) => {
                console.log(err.code, err.message);
                dispatch( finishLoading() );
                Swal.fire('Error', err.message, 'error')
            })

    }
}

export const startRegisterWithEmailPasswordName = (email, password, name) => {
    return (dispatch) => {
        const auth = getAuth();

        createUserWithEmailAndPassword(auth, email, password)
            .then( async ({ user }) => {

                await updateProfile(auth.currentUser, {
                    displayName: name,  photoURL: ""
                })

                dispatch(login(user.uid, user.displayName, 'firebase', ''))

            })
            .catch( (err) => {
                console.log(err.code, err.message);
                Swal.fire('Error', err.message, 'error')
            })

        console.log(auth.currentUser)
    }
}

export const startGoogleLogin = () =>{
    return (dispatch) =>{
        const auth = getAuth();
        signInWithPopup(auth, googleAuthProvider)
            .then( res => {
            })
    }
}

export const login = (uid, name, type, token) => ({
    type: types.login,
    payload: {
        uid,
        name,
        type,
        token
    }
})

export const startLogout = () => {
    return async (dispatch) => {
        const auth = getAuth();
        await signOut(auth)
        dispatch( logout() );
    }
}

export const logout = () => ({
    type: types.logout
})
