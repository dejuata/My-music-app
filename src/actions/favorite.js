import { types } from "../types/types";

export const setFavorites = (payload) => ({
    type: types.setFavorites,
    payload
})

export const removeFavorite= (payload) => ({
    type: types.removeFavorite,
    payload
})

export const logoutFavorites = () => ({
    type: types.logoutFavorite
})