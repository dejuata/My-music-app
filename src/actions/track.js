import { types } from "../types/types";

export const setTracks = (tracks) => ({
    type: types.setTracks,
    payload: tracks
})

export const addTrackFavorite = (id) => ({
    type: types.addTrackFavorite,
    payload: id
})

export const removeTrackFavorite = (id) => ({
    type: types.removeTrackFavorite,
    payload: id
})

export const logoutTrack = () => ({
    type: types.logoutTrack
})