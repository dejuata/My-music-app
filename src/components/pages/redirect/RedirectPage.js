import React, { useEffect } from 'react'
import { useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { login, setToken } from '../../../actions/auth';

import { getTokenFromUrl } from '../../../api/requests/getTokenFromUrl';
import { getUserInfo } from '../../../api/requests/getUserInfo';

export const RedirectPage = () => {

    const { hash } = useLocation();
    const dispatch = useDispatch();

    useEffect(() => {
        const _token = getTokenFromUrl(hash);

        dispatch( setToken(_token) );

        getUserInfo()
            .then( ( {id, displayName} ) => {
                dispatch( login(id, displayName, 'spotify', _token) );
            })

    }, [dispatch, hash])

    return (
        <>
        </>
    )
}
