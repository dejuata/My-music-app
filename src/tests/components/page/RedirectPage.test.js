import { RedirectPage } from "../../../components/pages/redirect/RedirectPage"
import { mount } from 'enzyme';
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

import configureStore from 'redux-mock-store';
import thunk from "redux-thunk";
import { setToken } from "../../../actions/auth";


jest.mock("../../../actions/auth", () => ({
    setToken: jest.fn(),
}))

const middlewares = [ thunk ];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        token: null
    }
};

let store = mockStore(initState);
store.dispatch = jest.fn()

const wrapper = mount(
    <Provider store={ store }>
        <MemoryRouter>
            <RedirectPage />
        </MemoryRouter>
    </Provider>
)

describe('Test RedirectPage', () => {

    test('Should be render RedirectPage', () => {

        expect(wrapper).toMatchSnapshot();

    })

})