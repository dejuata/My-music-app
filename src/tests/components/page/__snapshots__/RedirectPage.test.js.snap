// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RedirectPage Should be render RedirectPage 1`] = `""`;
